src/ii_collections/n19Sum.kt

```diff
@@ -7,5 +7,5 @@ fun example6() {
 fun Customer.getTotalOrderPrice(): Double {
     // Return the sum of prices of all products that a customer has ordered.
     // Note: a customer may order the same product for several times.
-    return orders.flatMap { it.products }.map { it.price }.sum()
+    return orders.flatMap { it.products }.sumByDouble { it.price }
```

Comment:
@ranjeevmahtani I am just contributer like you, so I won't decide. I can only say that your solution is really interesting, but as `kotlin-koans` is dedicated for beginners, this solution might be a bit tough to guess for guys who don't know kotlin syntax sugar as good as you.